js/main.js

```diff
@@ -40,6 +39,20 @@ document.addEventListener('DOMContentLoaded', function (e) {
   }
 });
 
+document.addEventListener('click', function (e) {
+  if (!e.target.hasAttribute('href')) {
+    return;
+  }
+
+  var editProfileForm = document.querySelector('main > div[data-view]');
+  if ($form.elements.username.value.trim() === '' && editProfileForm.className === 'show') {
```

Comment:
This should be checking if the username stored in the `data.profile` object is empty, not the form's input value.